adapter/config/crd/client.go

```diff
@@ -145,7 +145,7 @@ func (cl *Client) RegisterResources() error {
 	}
 
 	for _, schema := range cl.descriptor {
-		name := schema.Plural + "." + model.IstioAPIGroup
+		name := resourceName(schema.Plural) + "." + model.IstioAPIGroup
```

Comment:
Why not change `schema.Plural` to be the non-dashed version? Is this to avoid making changes to istio/istio configuration artifacts, and if so, would this lead to some user confusion around where we need to use `route-rules` vs. `routerules` as the type?

Reply:
k8s spec says the name should a DNS name and dash is allowed to separate words. We need a word separator to get "RouteRule" instead of "Routerule".

Reply:
How is this different than `CustomResourceDefinitions`/`customresourcedefinitions`?

Reply:
k8s uses capitalized form as canonical (CustomResourceDefinition) and does the reverse translation to dashed names somewhere. We can do the same, but that breaks the existing documentation and examples for `istioctl`.

Reply:
Ya, that seems to be the tradeoff: make everything consistent now between proxy, mixer, and k8s style and deal with the upfront migration cost, or make this simpler change and live with inconsistent types. We have an opportunity to fix this now since there are a lot of other breaking config/doc changes that need to be made in the next few weeks. Once we're past 0.2.x release it will be much harder to fix this (if its actually a problem).

Reply:
The dashes are only used by istioctl, right? So we're fully consistent with k8s as long as kubectl is used.

Reply:
"make everything consistent now between proxy, mixer, and k8s style and deal with the upfront migration cost" seems right to me, before the first 0.2